app/ssttpaffordability/AffordabilityController.scala

```diff
@@ -88,20 +88,32 @@ class AffordabilityController @Inject() (
       val spending = journey.maybeSpending.fold(Seq.empty[Expenses])(_.expenses)
       val income = journey.maybeIncome.fold(Seq.empty[IncomeBudgetLine])(_.budgetLines)
       if (spending.nonEmpty && income.nonEmpty) {
-        Redirect(ssttpaffordability.routes.AffordabilityController.getHowMuchYouCouldAfford())
+        Redirect(ssttpaffordability.routes.AffordabilityController.saveTotalLeftOverIncome())
       } else {
         Future.successful(Ok(views.add_income_spending(income, spending)))
       }
     }
   }
 
+  def saveTotalLeftOverIncome: Action[AnyContent] = as.authorisedSaUser.async { implicit request =>
+    JourneyLogger.info(s"AffordabilityController.getHowMuchYouCouldAfford: $request")
+    journeyService.authorizedForSsttp { journey =>
+      val spending = journey.maybeSpending.fold(Seq.empty[Expenses])(_.expenses)
+      val income = journey.maybeIncome.fold(Seq.empty[IncomeBudgetLine])(_.budgetLines)
+      val totalLeftOverIncome = income.map(_.amount).sum - spending.map(_.amount).sum
```

Comment:
Can this now be  `val totalLeftOverIncome = journey.totalLeftOverIncome` ? (keeping the calculation in one place )

Reply:
Kostas has helped me with a solution